yarn-project/circuit-types/src/interfaces/block-prover.ts

```diff
@@ -0,0 +1,26 @@
+import { Fr, GlobalVariables, Proof } from "@aztec/circuits.js";
+import { L2Block } from "../l2_block.js";
+import { ProcessedTx } from "../index.js";
+
+export type ProvingResult = {
+  block: L2Block;
+  proof: Proof;
+}
+
+/**
+ * The interface to the prover client.
+ * Provides the ability to generate proofs and build rollups.
+ */
+export interface BlockProver {
```

Comment:
I'm a bit confused by this interface. why does `proveBlock` accept `txs` if there is an `addNewTx` function? Also, it looks like both the `TxProver` and `ProvingOrchestrator` have separate (but similar) implementations of `proveBlock`. 

I'd also expect `proveBlock` to return `Promise<ProvingResult>`.

I'm also surprised that `startNewBlock` accepts a callback function: why not return a `Promise<ProvingResult>`? 

Further, how does this interface with the `BlockBuilder` interface? I can't clearly see how these pieces fit, and what implementation *"is"* what.